### PR TITLE
Add PersonalSections Controller Delete and Tests

### LINKED ISSUE #26: FEATURE: I can delete courses from a personal schedule on the main personal schedule page BACKEND
Acceptance Criteria:
- [x] Create DELETE endpoint in PersonalSectionsController with endpoint DELETE /api/personalSections/delete
- [x] Add tests for the new controller method

Functionality:
- The endpoint should remove all lectures and sections associated with psId and enrollCd
- Tests will ensure 100% coverage and mutation

Rationale:
- This endpoint is necessary as the personal sections table does not include the id variable necessary for users to delete specific courses through the existing API at DELETE /api/courses/user

### PR DESCRIPTION
**Pull request by WinbertZhang**
Merge Order:
1. #28 
2. #29 
3. #24 

This PR adds a backend API with endpoint /api/personalSections/delete. It takes in psId and enrollmentCd and removes all related lectures and sections for a course. Adds tests that achieve 100% mutation coverage. This PR is necessary to enable #29 to delete a course from the Personal Sections table as the current courses DELETE API requires an id variable not present in the personal sections GET API.

DOKKU DEPLOYMENT: https://team04-winbertzhang-dev.dokku-04.cs.ucsb.edu/

Steps:
Login, then create a Personal Schedule at https://team04-winbertzhang-dev.dokku-04.cs.ucsb.edu/personalschedules/list

Add a course to the personal schedule at https://team04-winbertzhang-dev.dokku-04.cs.ucsb.edu/

Open the Swagger UI and find the DELETE /api/personalSections/delete route:
https://team04-winbertzhang-dev.dokku-04.cs.ucsb.edu/swagger-ui/index.html

Enter the psId and enrollCd of the newly added course:

Screenshot 2024-05-26 at 5 11 30 PM

Execute the command and you should receive a status 200 code with response message Schedule with psId and associated lectures with enrollCd deleted:

Screenshot 2024-05-26 at 5 13 11 PM

Closes #26

### CODE REVIEW

**Review by emrecikisir — APPROVED** [review]
LGTM

**Review by andrew344c — APPROVED** [review]
LGTM

**Review by whuang602 — CHANGES_REQUESTED** [review]
Could you include a deployment for this?

**Comment by pconrad** [comment]
@WinbertZhang I asked for a deploy link two days ago.  No progress, so marking it Stalled.

**Comment by pconrad** [comment]
> @WinbertZhang I asked for a deploy link two days ago. No progress, so marking it Stalled.

Can we use the deployment for PR29 to test this?  It appears that we can, but I want to stress that it's so helpful if you can point this out for the reviewers/testers rather than making us figure it out on our own.

When I say helpful, I mean "to me", yes, but mostly I mean: "to you, your team, and the whole class".

If I can get through a PR in 5 minutes, then I can review every teams queue every 24 hours, and still have time left to help folks with individual issues.  

When they take 10-15 minutes, it takes me 48-72 hours to get through the queue, and I have less time (or no time) to help folks with individual issues.  Please help.

**Comment by WinbertZhang** [comment]
> > @WinbertZhang I asked for a deploy link two days ago. No progress, so marking it Stalled.
> 
> Can we use the deployment for PR29 to test this? It appears that we can, but I want to stress that it's so helpful if you can point this out for the reviewers/testers rather than making us figure it out on our own.
> 
> When I say helpful, I mean "to me", yes, but mostly I mean: "to you, your team, and the whole class".
> 
> If I can get through a PR in 5 minutes, then I can review every teams queue every 24 hours, and still have time left to help folks with individual issues.
> 
> When they take 10-15 minutes, it takes me 48-72 hours to get through the queue, and I have less time (or no time) to help folks with individual issues. Please help.

Sorry for the confusion. I was holding off on posting the deployment URL until PR 34 was merged to allow for the Swagger UI to be present in the deployed app to make it easier to test this PR. That's why I marked this PR as on hold waiting for another PR to merge. I'll make sure to write that in the comments next time to avoid any confusion.

Here is the updated deployment URL: https://team04-winbertzhang-dev.dokku-04.cs.ucsb.edu/

**Comment by WinbertZhang** [comment]
PR Merge order should be:
1. #28 
2. #29 
3. #24

File: src/main/java/edu/ucsb/cs156/courses/controllers/PersonalSectionsController.java
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/courses/controllers/PersonalSectionsController.java:105]: Can we move this code to the existing PSCourseController.java instead of putting it here?

And let's add another story to move the other api endpoint over into PSCourseController as well.  This controller should probably not have been built in the first place as a separate file.

Comment (pconrad) [reply src/main/java/edu/ucsb/cs156/courses/controllers/PersonalSectionsController.java:105]: Ok, fine, since we are close to the deadline, I'll let it go.. but I hate the fact that we have two controller files and two controller tests where we should have one.

**Review by whuang602 — APPROVED** [review]
LGTM

**Review by pconrad — APPROVED** [review]
OK, LGTM.  I had to make my peace with the duplicate controllers for PSCourse, but I realize that may have happened at a much earlier stage.

**Pull request merged**
